organisaatio-ui/src/components/Sivut/Ryhmat/Muokkaus/MuokkausLomake.tsx

```diff
@@ -85,47 +86,27 @@ const MuokkausLomake = ({
                     <div className={styles.Rivi}>
                         <div className={styles.Kentta}>
                             <label>{i18n.translate('RYHMAN_NIMI')}</label>
-                            <div className={styles.Rivi}>
-                                <label>{i18n.translate('SUOMEKSI')}</label>
-                                <div className={styles.PitkaInput}>
-                                    <Input {...nimiFiBind} />
-                                </div>
-                            </div>
-                            <div className={styles.Rivi}>
-                                <label>{i18n.translate('RUOTSIKSI')}</label>
-                                <div className={styles.PitkaInput}>
-                                    <Input {...nimiSvBind} />
+                            {[nimiFiBind, nimiSvBind, nimiEnBind].map((bind, index) => (
+                                <div className={styles.Rivi}>
+                                    <label>{i18n.translate(KIELEKSI[index])}</label>
+                                    <div className={styles.PitkaInput}>
+                                        <Input {...bind} />
+                                    </div>
                                 </div>
-                            </div>
-                            <div className={styles.Rivi}>
-                                <label>{i18n.translate('ENGLANNIKSI')}</label>
-                                <div className={styles.PitkaInput}>
-                                    <Input {...nimiEnBind} />
-                                </div>
-                            </div>
+                            ))}
                         </div>
                     </div>
                     <div className={styles.Rivi}>
                         <div className={styles.Kentta}>
                             <label>{i18n.translate('RYHMAN_KUVAUS')}</label>
-                            <div className={styles.Rivi}>
-                                <label>{i18n.translate('SUOMEKSI')}</label>
-                                <div className={styles.PitkaInput}>
-                                    <Input {...kuvaus2FiBind} />
-                                </div>
-                            </div>
-                            <div className={styles.Rivi}>
-                                <label>{i18n.translate('RUOTSIKSI')}</label>
-                                <div className={styles.PitkaInput}>
-                                    <Input {...kuvaus2SvBind} />
-                                </div>
-                            </div>
-                            <div className={styles.Rivi}>
-                                <label>{i18n.translate('ENGLANNIKSI')}</label>
-                                <div className={styles.PitkaInput}>
-                                    <Input {...kuvaus2EnBind} />
+                            {[kuvaus2FiBind, kuvaus2SvBind, kuvaus2EnBind].map((bind, index) => (
```

Comment:
much better, but: this seems very brittle. what if localization key would be attribute of LanguagedInputBind? no need to try to juggle two separate arrays etc. Also: remember to add "key" -property to top level div, it should be there for a reason.
Actually whole thing within the loop should be separate component

Reply:
would it make sense to pass different LanguagedInputBind:s as array `{ nimi: LanguagedInputBind[] }`?
I suppose it would make Props easier to grasp? (also consider adding new languages to system...)